src/main/resources/referenceImageTemplateWithScript.json

```diff
@@ -105,7 +105,7 @@
                     "osDisk": {
                         "name": "osdisk",
                         "vhd": {
-                            "uri": "[concat('http://',variables('storageAccountName'),'.blob.core.windows.net/',variables('storageAccountContainerName'),'/', variables('vmName'), copyIndex(), 'OSDisk.vhd')]"
+                            "uri": "[concat('http://',variables('storageAccountName'),'.blob.core.windows.net/',variables('storageAccountContainerName'),'/', variables('vmName'), uniquestring(resourceGroup().id), 'OSDisk.vhd')]"
```

Comment:
Ahh, I guess because uniqueString replaces that.  Can you ensure that OS disk cleanup still works?  Ensure it's not making assumptions based on disk name.